File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.java
class Solution {
    public long pickGifts(int[] gifts, int k) {
        int n=gifts.length;
        
        while(k>0)
        {
            Arrays.sort(gifts);
            gifts[n-1]=(int)Math.sqrt(gifts[n-1]);
            
            k--;
        }
        long sum=0;
        for(int i=0;i<gifts.length;i++)
        {
            sum+=gifts[i];
        }
        return sum;
    }
}

File: 2558-take-gifts-from-the-richest-pile/2558-take-gifts-from-the-richest-pile.cpp
class Solution {
public:
    long long pickGifts(vector<int>& gifts, int k) {
        priority_queue<int> q;
        for(int i=0; i<gifts.size();i++)
        {
            q.push(gifts[i]);
        }
        while(k--)
        {
            int a=q.top();
            q.pop();
            a=sqrt(a);
            q.push(a);
        }
        long long result=0;
        while(!q.empty())
        {
            result+=q.top();
            q.pop();
        }
        return result;
    }
};